src/modules/producer.js

```diff
@@ -118,7 +119,7 @@ function checkRpc (queue, msg, options) {
 
         if (publishOrSendToQueue.call(this, queue, msg, options)) {
             //defered promise that will resolve when response is received
-            let responsePromise = Promise.defer();
+            let responsePromise =  new Deferred();
```

Comment:
Sure, it would take too long otherwise

For the refactor, you can also mention that the whole project needs a proper refactor using es6 syntax more globally - with proper classes, await, etc. and not usage of arrow functions and `this` everywhere - it's a real mess!